src/dfmt/formatter.d

```diff
@@ -726,7 +730,13 @@ private:
             {
                 indents.pop();
             }
-            indents.popTempIndents();
+            if (indents.length > 0 && indents.topIsTemp()) {
+                savedIndents ~= IndentStack(config);
+            }
+            while (indents.length > 0 && indents.topIsTemp()) {
+                savedIndents.back.push(indents.top, indents.topDetails);
```

Comment:
`.back.push`?

Reply:
A new element in the array is reserved in the condition before (`savedIndents ~= IndentStack(config);`) and that pushes the indentation to this element.